capstone-frontend/src/components/SearchUserModal.jsx

```diff
@@ -0,0 +1,255 @@
+import React, { Component } from 'react';
+
+import { Button, Form, Spinner, Modal, Col, Row, Card } from 'react-bootstrap'
+
+import '../styles/SearchUserModal.css'
+import '../styles/Modal.css'
+
+export class SearchUserModal extends Component {
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      fetchingUsers: false, // For Spinner
+      showModal: false,
+      typingTimeout: 0,
+      searchTerm: '',
+      searchResults: [],
+      addedUsersTracker: [],
```

Comment:
Just a couple of clarification questions here...
1. Is `addedUsersTracker` just a combination of `addedUsers` and `addedFriends` below? 
2. What is the distinction between an "addedUser" and an "addedFriend" ? When would a user choose to add a user vs. add a friend?

Reply:
The ```addedUsersTracker``` keeps track of Users added to ```addedUsers``` and addedFriends``` and displays them in the modal. ```addedUsersTracker``` basically combines both of these array into a set to avoids displaying duplicate data. 

```addedUsers``` is meant to be used as a helper array. For example, I will use it to keep track of Collaborators. I think @steven-solar will use it to keep track of the members of the club.

```addedFriends``` is meant to keep track of the User's friends. This allows an owner/admin to not only add a User to Club but also add them as a Friend. 

In the future, we might change the Friend functionality of the modal.